### PR TITLE
[ADD] base_import FAQ

### PR DESCRIPTION
**Pull request by csnauwaert**

### CODE REVIEW

**Comment by mart-e** [comment]
Name your files (either the filename or the folder) so it's clear it's for base_import doc page

File: general/base_import/import_faq.rst [hunk 1]
```diff
@@ -0,0 +1,201 @@
+=======================
+Import CSV file to Odoo
+=======================
+
+
+
+Frequently Asked Questions
+--------------------------
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: why is this empty? if you mean the points below are the questions, just remove it

File: general/base_import/import_faq.rst [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+
+Need to import data from an other application?
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: missing interrogative word. "How to import data from another application?" or something like that

File: general/base_import/import_faq.rst [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+----------------------------------------------
+
+In order to re-create relationships between different records, you should use the unique identifier from the original application and map it to the **ID** (External ID) column in Odoo. When you import an other record that links to the first one, use **XXX/ID** (XXX/External ID) to the original unique identifier.
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: You are not going to like it but keep the 80 char limit for your text. It's way easier when we will need to correct typos or change a few things in a long text block.

File: general/base_import/import_faq.rst [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+The **ID** (External ID) will also be used to update the original import if you need to re-import modified data later, it's thus good practice to specify it whenever possible.
+
+
+
+I cannot find the field I want to map my column to?
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: Adding a question mark does not make it a question. This sounds so frenchy I can smell the cheese ;-)

File: general/base_import/import_faq.rst [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+---------------------------------------------------
+
+Odoo try to find with some heuristic, based on the first ten lines of the files, the type of field for each columns inside your file. For example if you have a column only containing numbers, only the fields that are of type integer will be displayed for you to choose from. While this behaviour might be good and easy for most cases scenarios, it is also possible that it goes wrong sometimes or that you want to map your column to a field that is not proposed by default.
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: try -> tries, heuristic -> heuristics, of the files -> of the file

File: _static/example_files/purchase.order_functional_error_line_cant_adpat.csv [hunk 6]
```diff
@@ -0,0 +1,6 @@
+"Order Reference","Supplier","Destination","Pricelist","Order Lines / Product","Order Lines / Quantity"
```
Comment (rim-odoo) [inline _static/example_files/purchase.order_functional_error_line_cant_adpat.csv]: Maybe you want to rename your file into purchase.order_functional_error_line_cant_**adapt**.csv ;-)

File: general/base_import/import_faq.rst [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+If that happens, you just have to check the **Show all fields for completion (advanced)** option, you will then be able to choose from the complete list of fields for each columns.
```
Comment (rim-odoo) [inline general/base_import/import_faq.rst]: If that happens -> in this case?

File: general/base_import/import_faq.rst (hunk 4 above)
Comment (rim-odoo) [inline general/base_import/import_faq.rst]: How to map my columns to Odoo fields?

File: general/base_import/import_faq.rst [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+
+Where can I change the date import format?
+------------------------------------------
+
+Odoo can automatically detect if a column is a date and it will try to guess the date format from a set of most used date format. While this process can work for a lot of simple date format, some exotic date format will not be recognize and it is also possible to have some confusion (day and month inverted as example) as it is difficult to guess correctly which part is the day and which one is the month in a date like '01-03-2016'.
+
+To view which date format Odoo has found from your file you can check the **Date Format** that is shown when clicking on **Options** under the file selector. If this format is incorrect you can change it to your liking using the *ISO 8601* to define the format.
+
+.. note::
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: maybe use `.. tip::` for this

File: general/base_import/import_faq.rst [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+    If you are importing an excel (.xls, .xlsx) file, you can use date cells to store dates as the display of dates in excel is different from the way it is stored. That way you will be sure that the date format is correct in Odoo whatever your locale date format is.
+
+
+
+Can I import numbers with currency sign (e.g.: $32.00)?
+-------------------------------------------------------
+
+Yes, we fully support numbers with parenthesis to represent negative sign as well as numbers with currency sign attached to them. Odoo also automatically detect which thousand/decimal separator you use (you can change those under **options**). If you use a currency symbol that is not known to Odoo, it might not be recognized as a number though and it will crash.
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: avoid using "we", "you" when possible but try to use "it", "Odoo", "the user",...

File: general/base_import/import_faq.rst [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+- $ (32.000,00)
+
+
+
+What can I do when the Import preview table isn't displayed correctly?
+----------------------------------------------------------------------
+
+By default the Import preview is set on commas as field separators and quotation marks as text delimiters. If your csv file does not have these settings, you can modify the File Format Options (displayed under the Browse CSV file bar after you select your file).
+
+Note that if your CSV file has a tabulation as separator, Odoo will not detect the separations. You will need to change the file format options in your spreadsheet application. See the following question.
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: `.. note::`

File: general/base_import/import_faq.rst [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+
+- ABC 32.000,00
+- $ (32.000,00)
+
+
+
+What can I do when the Import preview table isn't displayed correctly?
+----------------------------------------------------------------------
+
+By default the Import preview is set on commas as field separators and quotation marks as text delimiters. If your csv file does not have these settings, you can modify the File Format Options (displayed under the Browse CSV file bar after you select your file).
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: maybe add a screeshot here (usually good idea when we need to describe the position of a button)

File: general/base_import/import_faq.rst [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+
+Note that if your CSV file has a tabulation as separator, Odoo will not detect the separations. You will need to change the file format options in your spreadsheet application. See the following question.
+
+
+
+How can I change the CSV file format options when saving in my spreadsheet application?
+---------------------------------------------------------------------------------------
+
+If you edit and save CSV files in speadsheet applications, your computer's regional settings will be applied for the separator and delimiter. We suggest you use OpenOffice or LibreOffice Calc as they will allow you to modify all three options (in 'Save As' dialog box > Check the box 'Edit filter settings' > Save).
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: Use the tag ```:menuselection:`First --> Second --> Third```` to represent steps

File: general/base_import/import_faq.rst [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+
+- Country: the name or code of the country
+- Country/Database ID: the unique Odoo ID for a record, defined by the ID postgresql column
+- Country/External ID: the ID of this record referenced in another application (or the .XML file that imported it) 
+
+For the country Belgium, you can use one of these 3 ways to import: 
+
+- Country: Belgium
+- Country/Database ID: 21
+- Country/External ID: base.be
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: We need some formatting here. Maybe the name of the column in bold.

File: general/base_import/import_faq.rst [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+If for example you have two product categories with the child name "Sellable" (ie. "Misc. Products/Sellable" & "Other Products/Sellable"), your validation is halted but you may still import your data. However, we recommend you do not import the data because they will all be linked to the first 'Sellable' category found in the Product Category list ("Misc. Products/Sellable"). We recommend you modify one of the duplicates' values or your product category hierarchy.
+
+However if you do not wish to change your configuration of product categories, we recommend you use make use of the external ID for this field 'Category'.
+
+
+
+How can I import a many2many relationship field (e.g. a customer that has multiple tags)?
+-----------------------------------------------------------------------------------------
+
+The tags should be separated by a comma without any spacing. For example, if you want your customer to be linked to both tags 'Manufacturer' and 'Retailer' then you will encode "Manufacturer,Retailer" in the same column of your CSV file.
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: Use escaping for the CSV part ``` ``Manufacturer,Retailer`` ``` to show it's exact text

File: general/base_import/import_faq.rst [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+What can I do if I have multiple matches for a field?
+-----------------------------------------------------
+
+If for example you have two product categories with the child name "Sellable" (ie. "Misc. Products/Sellable" & "Other Products/Sellable"), your validation is halted but you may still import your data. However, we recommend you do not import the data because they will all be linked to the first 'Sellable' category found in the Product Category list ("Misc. Products/Sellable"). We recommend you modify one of the duplicates' values or your product category hierarchy.
+
+However if you do not wish to change your configuration of product categories, we recommend you use make use of the external ID for this field 'Category'.
+
+
+
+How can I import a many2many relationship field (e.g. a customer that has multiple tags)?
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: Do not put the "e.g." in the title, way too long (but can be put in the sentence below)

File: general/base_import/import_faq.rst [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+
+`CSV file for Manufacturer, Retailer <../../_static/example_files/m2m_customers_tags.csv>`_.
+
+
+
+How can I import a one2many relationship (e.g. several Order Lines of a Sales Order)?
+-------------------------------------------------------------------------------------
+
+If you want to import sales order having several order lines; for each order line, you need to reserve a specific row in the CSV file. The first order line will be imported on the same row as the information relative to order. Any additional lines will need an addtional row that does not have any information in the fields relative to the order.
+As an example, here is purchase.order_functional_error_line_cant_adpat.CSV file of some quotations you can import, based on demo data.
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: Instead of "here is..." and link below, put it inline

File: general/base_import/import_faq.rst [hunk 17]
```diff
@@ -0,0 +1,201 @@
...
+The following CSV file shows how to import customers and their respective contacts:
+
+`Customers and their respective contacts <../../_static/example_files/o2m_customers_contacts.csv>`_.
+
+
+
+Can I import several times the same record?
+-------------------------------------------
+
+If you import a file that contains one of the column "External ID" or "Database ID", records that have already been imported will be modified instead of being created. This is very usefull as it allows you to import several times the same CSV file while having made some changes in between two imports. Odoo will take care of creating or modifying each record depending if it's new or not.
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: I think this is a very important part. I would rephrase it to say this is why you **should** put an external id when creating your CSV.

File: general/base_import/import_faq.rst [hunk 18]
```diff
@@ -0,0 +1,201 @@
...
+
+
+How to export/import different tables from an SQL application to Odoo?
+----------------------------------------------------------------------
+
+If you need to import data from different tables, you will have to recreate relations between records belonging to different tables. (e.g. if you import companies and persons, you will have to recreate the link between each person and the company they work for).
+
+To manage relations between tables, you can use the "External ID" facilities of Odoo. The "External ID" of a record is the unique identifier of this record in another application. This "External ID" must be unique accoss all the records of all objects, so it's a good practice to prefix this "External ID" with the name of the application or table. (like 'company_1', 'person_1' instead of '1')
+
+As an example, suppose you have a SQL database with two tables you want to import: companies and persons. Each person belong to one company, so you will have to recreate the link between a person and the company he work for. (If you want to test this example, here is a <a href="/base_import/static/csv/database_import_test.sql">dump of such a PostgreSQL database</a>)
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: the link to the SQL dump needs to be converted

File: general/base_import/import_faq.rst [hunk 19]
```diff
@@ -0,0 +1,201 @@
...
+
+We will first export all companies and their "External ID". In PSQL, write the following command:
```
Comment (mart-e) [inline general/base_import/import_faq.rst]: Put a link on the "In PSQL" to point to http://www.postgresql.org/docs/9.5/static/app-psql.html
